<launch>
  <arg name="cam_frame" default="camera_color_optical_frame"/>
  <arg name="cam_ref_frame" default="campero_base_footprint"/>
  <arg name="robot_frame" default="base_link"/>
  <arg name="max_dist_error" default="7"/> <!-- distancia de error maximo en pixeles entre el punto real y el punto en la img-->
  <arg name="canny_th1" default="100"/>
  <arg name="canny_th2" default="300"/>
  <arg name="blur_ksize" default="3"/>
  <arg name="erode_type" default="0"/>
  <arg name="erode_size" default="3"/>
  <arg name="dilate_type" default="0"/>
  <arg name="dilate_size" default="3"/>

  <node name="image_inpainting" pkg="image_inpainting"
      type="image_inpainting" respawn="false"
      output="screen">
      <param name="camera_frame"       value="$(arg cam_frame)"/>
      <param name="cam_ref_frame"    value="$(arg robot_frame)"/>
      <param name="robot_frame"       value="$(arg robot_frame)"/>
      <param name="max_dist_error"       value="$(arg max_dist_error)"/>
      <param name="canny_th1"       value="$(arg canny_th1)"/>
      <param name="canny_th2"       value="$(arg canny_th2)"/>
      <param name="blur_ksize"       value="$(arg blur_ksize)"/>
      <param name="erode_type"       value="$(arg erode_type)"/>
      <param name="erode_size"       value="$(arg erode_size)"/>
      <param name="dilate_type"       value="$(arg dilate_type)"/>
      <param name="dilate_size"       value="$(arg dilate_size)"/>
  </node>

</launch>
